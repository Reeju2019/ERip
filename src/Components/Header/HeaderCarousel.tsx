import * as React from 'react';
import { Carousel, Container, Image } from 'react-bootstrap';
import './Header.css';

interface IHeaderCarouselProps {}

const HeaderCarousel: React.FunctionComponent<IHeaderCarouselProps> = (
    props
) => {
    return (
        <>
            <Container fluid className="carousel-container-1 mt-2">
                <div className="row carousel-content mx-4">
                    <div
                        id="carouselExampleIndicators"
                        className="carousel slide"
                        data-bs-ride="true"
                    >
                        <Carousel>
                            <Carousel.Item>
                                <div className="d-flex flex-row">
                                    <div className="w-100 d-flex justify-content-center align-item-center align-self-center">
                                        <h2 className="txt col-8 justify-content-start text-white fs-2">
                                            Repair & Maintainence <br />
                                            Instant Doorstep Services On Demand
                                        </h2>
                                    </div>
                                    <Image
                                        className="d-flex caro-img justify-content-center"
                                        src="/static/image/carousel-img-1.png"
                                        alt="First slide"
                                    />
                                </div>
                            </Carousel.Item>
                            <Carousel.Item>
                                <div className="d-flex flex-row">
                                    <div className="w-100 d-flex justify-content-center align-item-center align-self-center">
                                        <h2 className="txt col-8 justify-content-start text-white fs-2">
                                            Top Trained Experts <br/> at Your Doorstep
                                    Right Away!
                                        </h2>
                                    </div>
                                    <Image
                                        className="d-flex caro-img justify-content-center"
                                        src="/static/image/carousel-img-1.png"
                                        alt="First slide"
                                    />
                                </div>
                            </Carousel.Item>
                            <Carousel.Item>
                                <div className="d-flex flex-row">
                                    <div className="w-100 d-flex justify-content-center align-item-center align-self-center">
                                        <h2 className="txt col-8 justify-content-start text-white fs-2">
                                            Easy Fix <br/> Superior Service Quality Parts
                                        </h2>
                                    </div>
                                    <Image
                                        className="d-flex caro-img justify-content-center"
                                        src="/static/image/carousel-img-1.png"
                                        alt="First slide"
                                    />
                                </div>
                            </Carousel.Item>
                        </Carousel>
                    </div>
                </div>
            </Container>
        </>
    );
};

export default HeaderCarousel;
