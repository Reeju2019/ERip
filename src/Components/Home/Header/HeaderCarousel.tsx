import { Carousel, Container, Image } from 'react-bootstrap'
import './Header.css'
import HeaderData from '../../../Data/Header.mock.json'

const HeaderCarousel: React.FunctionComponent = () => {
  return (
    <>
      <Container fluid className='carousel-container-1 mt-2'>
        <div className='row carousel-content mx-4'>
          <div id='carouselExampleIndicators' className='carousel slide' data-bs-ride='true'>
            <Carousel>
              <Carousel.Item>
                <div className='d-flex flex-row'>
                  <div className='w-100 d-flex justify-content-center align-item-center align-self-center'>
                    <h2 className='txt col-8 justify-content-start text-white fs-2'>
                      {HeaderData.header_caro_1_1}
                      <br />
                      {HeaderData.header_caro_1_2}
                    </h2>
                  </div>
                  <Image
                    className='d-flex caro-img justify-content-center'
                    src='/static/image/carousel-img-1.png'
                    alt='First slide'
                  />
                </div>
              </Carousel.Item>
              <Carousel.Item>
                <div className='d-flex flex-row'>
                  <div className='w-100 d-flex justify-content-center align-item-center align-self-center'>
                    <h2 className='txt col-8 justify-content-start text-white fs-2'>
                      {HeaderData.header_caro_2_1} <br />
                      {HeaderData.header_caro_2_2}
                    </h2>
                  </div>
                  <Image
                    className='d-flex caro-img justify-content-center'
                    src='/static/image/carousel-img-2.png'
                    alt='First slide'
                  />
                </div>
              </Carousel.Item>
              <Carousel.Item>
                <div className='d-flex flex-row'>
                  <div className='w-100 d-flex justify-content-center align-item-center align-self-center'>
                    <h2 className='txt col-8 justify-content-start text-white fs-2'>
                      {HeaderData.header_caro_3_1} <br />
                      {HeaderData.header_caro_3_2}
                    </h2>
                  </div>
                  <Image
                    className='d-flex caro-img justify-content-center'
                    src='/static/image/carousel-img-3.png'
                    alt='First slide'
                  />
                </div>
              </Carousel.Item>
            </Carousel>
          </div>
        </div>
      </Container>
    </>
  )
}

export default HeaderCarousel
