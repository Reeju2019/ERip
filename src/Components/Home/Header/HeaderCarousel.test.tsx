import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import HeaderCarousel from './HeaderCarousel'

test('header carousel', () => {
  render(<HeaderCarousel />)
})

describe('pnf rendering specification', () => {
  it('SnapShot testing', () => {
    const component = renderer.create(<HeaderCarousel />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('render without crashing', () => {
    const div = document.createElement('div')
    render(<HeaderCarousel />)
    ReactDOM.unmountComponentAtNode(div)
  })
})