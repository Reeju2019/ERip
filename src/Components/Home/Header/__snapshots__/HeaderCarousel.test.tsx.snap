// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pnf rendering specification SnapShot testing 1`] = `
<div
  className="carousel-container-1 g-0 container-fluid"
  style={
    Object {
      "backgroundImage": "url(static/image/carousel-bg-1.png)",
    }
  }
>
  <div
    className="row carousel-content p-0 m-0 g-0"
  >
    <div
      className="carousel slide"
      data-bs-ride="true"
      id="carouselExampleIndicators"
    >
      <div
        className="carousel slide"
        onKeyDown={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="carousel-indicators"
        >
          <button
            aria-current={true}
            aria-label="Slide 1"
            className="active"
            data-bs-target=""
            onClick={[Function]}
            type="button"
          />
          <button
            aria-current={false}
            aria-label="Slide 2"
            data-bs-target=""
            onClick={[Function]}
            type="button"
          />
          <button
            aria-current={false}
            aria-label="Slide 3"
            data-bs-target=""
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="carousel-inner"
        >
          <div
            className="active carousel-item"
            style={
              Object {
                "backgroundImage": "url(static/image/carousel-bg-1.png)",
              }
            }
          >
            <div
              className="d-flex flex-row"
            >
              <div
                className="w-100 d-flex justify-content-center align-item-center align-self-center"
              >
                <h2
                  className="txt col-8 justify-content-start text-white fs-2"
                >
                  Repair & Maintainence
                  <br />
                  Instant Doorstep Services On Demand
                </h2>
              </div>
              <img
                alt="First slide"
                className="d-flex caro-img justify-content-center"
                src="/static/image/carousel-img-1.png"
              />
            </div>
          </div>
          <div
            className="carousel-item"
            style={
              Object {
                "backgroundImage": "url(static/image/carousel-bg-1.png)",
              }
            }
          >
            <div
              className="d-flex flex-row"
            >
              <div
                className="w-100 d-flex justify-content-center align-item-center align-self-center"
              >
                <h2
                  className="txt col-8 justify-content-start text-white fs-2"
                >
                  Top Trained Experts
                   
                  <br />
                  at Your Doorstep Right Away!
                </h2>
              </div>
              <img
                alt="First slide"
                className="d-flex caro-img justify-content-center"
                src="/static/image/carousel-img-2.png"
              />
            </div>
          </div>
          <div
            className="carousel-item"
            style={
              Object {
                "backgroundImage": "url(static/image/carousel-bg-1.png)",
              }
            }
          >
            <div
              className="d-flex flex-row"
            >
              <div
                className="w-100 d-flex justify-content-center align-item-center align-self-center"
              >
                <h2
                  className="txt col-8 justify-content-start text-white fs-2"
                >
                  Easy Fix
                   
                  <br />
                  Superior Service Quality Parts
                </h2>
              </div>
              <img
                alt="First slide"
                className="d-flex caro-img justify-content-center"
                src="/static/image/carousel-img-3.png"
              />
            </div>
          </div>
        </div>
        <a
          className="carousel-control-prev"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            aria-hidden="true"
            className="carousel-control-prev-icon"
          />
          <span
            className="visually-hidden"
          >
            Previous
          </span>
        </a>
        <a
          className="carousel-control-next"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            aria-hidden="true"
            className="carousel-control-next-icon"
          />
          <span
            className="visually-hidden"
          >
            Next
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
